src/charm.py

```diff
@@ -93,6 +93,7 @@ def __init__(self, *args):
             charm=self,
             relationship_name=TLS_RELATION_NAME,
             certificate_requests=[self._get_certificate_request()],
+            refresh_events=[self.on.upgrade_charm],
```

Comment:
I think we should make the change in the tls lib instead of in every charm

Reply:
I agree.